source/main/network/Network.cpp

```diff
@@ -545,14 +569,23 @@ bool Connect()
 
 void Disconnect()
 {
-	m_shutdown = true;
-	m_send_packet_available_cv.notify_one();
-	m_send_thread.join();
-	m_recv_thread.detach();
+    LOG("[RoR|Networking] Disconnect() disconnecting...");
 
-	SendMessage(MSG2_USER_LEAVE, 0, 0, 0);
-	socket.set_timeout(1, 1000);
-	socket.disconnect();
+    m_shutdown = true; // Instruct Send/Recv threads to shut down.
+
+    m_send_packet_available_cv.notify_one();
+    m_send_thread.join();
+    LOG("[RoR|Networking] Disconnect() sender thread stopped...");
+
+    SendMessage(MSG2_USER_LEAVE, 0, 0, 0);
+    socket.set_timeout(1, 1000);
+    
+    m_recv_thread.join();
```

Comment:
Is there a reason why you don't join the receive thread right after you join the send thread?